src/ontoportal_client/api.py

```diff
@@ -110,7 +110,12 @@ def search(self, text: str, ontology: Optional[str] = None):
         params = {"q": text, "include": ["prefLabel"]}
         if ontology:
             params["ontologies"] = ontology
-        return self.get_json("/search", params)
+        page = "1"
+        while page:
+            params["page"] = page
+            result = self.get_json("/search", params)
+            yield result
+            page = result["nextPage"]
```

Comment:
this seems to work but the test doesn't represent what happens on the last page. Is there a `None` for `result["nextPage"]`? or does it blow up? we need a test for this

Reply:
Yes, `result["nextPage"]` will be `None` on the last page. Do you think it would be better to be more defensive and use `result.get("nextPage", None)` just in case?

The test I added called [`test_search_pagination`](https://github.com/pkalita-lbl/ontoportal-client/blob/a855f53f822db30443508ab61f21a36be70c00fe/tests/test_client.py#L32) that was intended to verify that the generator can be exhausted (and therefore the last page reached) without anything bad happening. Did you have something else in mind?

Reply:
Alright thanks for clarifying, I simplified the implementation a bit and added some docs. I'm pretty confident in this now.

Reply:
Nice, I like your changes. Thanks!